### PR TITLE
Bleve search experiment

### PR DESCRIPTION
**Pull request by timburks**
This cleans up and pulls together two experimental subcommands that index documents and query a search index built with https://blevesearch.com. Bleve appears to require local file storage, which makes it a bit unclear to see how we might easily use it in a service, but it is feature-rich and even with this simple setup, it produces useful ranked and scored results.

A test bleve index of ~4500 specs is around 4GB in size and queries generally complete in a few hundred milliseconds.

### CODE REVIEW

File: cmd/registry-experimental/cmd/bleve/search.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	_ "github.com/blevesearch/bleve/search/highlight/highlighter/ansi"
+	"github.com/spf13/cobra"
+)
+
+var number int
+var output string
+
+func searchCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "search",
```
Comment (theganyo) [inline cmd/registry-experimental/cmd/bleve/search.go:32]: "seach QUERY"

File: cmd/registry-experimental/cmd/bleve/search.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+					fmt.Fprintf(cmd.OutOrStdout(), ">> %d (%0.2f) %s\n", i+1, hit.Score, reduce(hit.ID))
+				}
+			default:
+				return fmt.Errorf("unknown output format: %q", output)
+			}
+			return nil
+		},
+	}
+	cmd.Flags().StringVarP(&output, "output", "o", "text", "output type (text|json|name)")
+	cmd.Flags().IntVarP(&number, "number", "n", 10, "number of results")
```
Comment (theganyo) [inline cmd/registry-experimental/cmd/bleve/search.go:84]: Is this a "limit"?

File: cmd/registry-experimental/cmd/bleve/search.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	return cmd
+}
+
+func reduce(name string) string {
```
Comment (theganyo) [inline cmd/registry-experimental/cmd/bleve/search.go:88]: Could you add a comment or rename? I'm still waking up and it took me too long to figure out what this was supposed to do. :)

**Review by theganyo — APPROVED** [review]

File: cmd/registry-experimental/cmd/bleve/search.go (hunk 1 above)
Comment (timburks) [reply cmd/registry-experimental/cmd/bleve/search.go]: thanks! fixed.

File: cmd/registry-experimental/cmd/bleve/search.go (hunk 2 above)
Comment (timburks) [reply cmd/registry-experimental/cmd/bleve/search.go]: Yes, I changed it to "limit" -- unfortunately the Registry API uses "page_size" (from the AIPs) so we still don't have one universal flag for this.

File: cmd/registry-experimental/cmd/bleve/search.go (hunk 3 above)
Comment (timburks) [reply cmd/registry-experimental/cmd/bleve/search.go:89]: I added a comment, I expect this would be revisited if we go further with this.

**Pull request merged**